roles/openshift_master/tasks/main.yml

```diff
@@ -23,8 +23,9 @@
   lineinfile:
     dest: /etc/sysconfig/openshift-master
     regexp: '^OPTIONS='
-    line: "OPTIONS=\"--public-master={{ openshift_public_ip }} --nodes={{ openshift_node_ips
-          | join(',') }}  --loglevel={{ openshift_master_debug_level }}\""
+    line: "OPTIONS=\"--public-master={{ openshift_public_ip if
+          openshift_hostname_workaround else ansible_fqdn  }} --nodes={{ openshift_node_ips
+              | join(',') }}  --loglevel={{ openshift_master_debug_level }}\""
```

Comment:
@detiber I hate doing complex logic like this where the task is being called because it's very hard to debug when the logic fails.

Consider doing this filter expression on a set_fact (possibly using an ansible conditional), and then using the fact that was set inside of this expression. That way we can use 'debug: var=newfact' in between the two to see what values we're getting when we need to debug it.

I won't block the PR on this, it's just a preference of mine. Let me know if you agree or not. If you want the PR merged as is, I'll do that.